src/main/java/jenkins/scm/api/SCMEvent.java

```diff
@@ -210,13 +211,41 @@ protected SCMEvent(SCMEvent<P> copy) {
     @NonNull
     protected static synchronized ScheduledExecutorService executorService() {
         if (executorService == null) {
-            // corePoolSize is set to 10, but will only be created if needed.
-            // ScheduledThreadPoolExecutor "acts as a fixed-sized pool using corePoolSize threads"
-            executorService = new ImpersonatingScheduledExecutorService(new ScheduledThreadPoolExecutor(10, new NamingThreadFactory(new ClassLoaderSanityThreadFactory(new DaemonThreadFactory()), "SCMEvent")), ACL.SYSTEM);
+            threadPoolExecutor = new ScheduledThreadPoolExecutor(
+                10,
+                new NamingThreadFactory(
+                    new ClassLoaderSanityThreadFactory(new DaemonThreadFactory()), "SCMEvent")
+            );
+            executorService = new ImpersonatingScheduledExecutorService(
+                threadPoolExecutor,
+                ACL.SYSTEM2);
         }
+
         return executorService;
     }
 
+    public static EventQueueMetrics getEventProcessingMetrics() {
+        return new EventQueueMetrics();
+    }
+
+    public static class EventQueueMetrics {
+        public int getPoolSize() {
+            return threadPoolExecutor == null ? 0 : threadPoolExecutor.getPoolSize();
```

Comment:
lazily initialised and calculated as required due to otel treating each resource as completely independent, originally I set all the fields in the constructor but that required fetching the data a lot more than required (some of the calls acquire locks)